pkg/controller/dataimportcron-controller_test.go

```diff
@@ -0,0 +1,315 @@
+
+				_, err = reconciler.Reconcile(context.TODO(), cronReq)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = reconciler.client.Get(context.TODO(), cronJobKey(cron), cronjob)
+				Expect(err).To(HaveOccurred())
+			}
+		},
+			Entry("create CronJob and add finalizer on a valid DataImportCron with source registry URL", false),
+			Entry("delete CronJob when DataImportCron with source registry URL has DeletionTimestamp", true),
+		)
+
+		DescribeTable("Should", func(setSucceeded bool) {
+			cron := newDataImportCron(cronName)
+			reconciler = createDataImportCronReconciler(cron)
+			_, err := reconciler.Reconcile(context.TODO(), cronReq)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = reconciler.client.Get(context.TODO(), cronKey, cron)
+			Expect(err).ToNot(HaveOccurred())
+
+			cron.Annotations[AnnSourceDesiredDigest] = testDigest
+			err = reconciler.client.Update(context.TODO(), cron)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = reconciler.Reconcile(context.TODO(), cronReq)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = reconciler.client.Get(context.TODO(), cronKey, cron)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cron.Status.CurrentImports).ToNot(BeNil())
+
+			dvName := cron.Status.CurrentImports[0].DataVolumeName
+			Expect(dvName).ToNot(BeEmpty())
+			digest := cron.Status.CurrentImports[0].Digest
+			Expect(digest).ToNot(BeEmpty())
+
+			dv := &cdiv1.DataVolume{}
+			err = reconciler.client.Get(context.TODO(), dvKey(dvName), dv)
+			Expect(err).ToNot(HaveOccurred())
+
+			if setSucceeded {
+				_, err = reconciler.Reconcile(context.TODO(), cronReq)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = reconciler.client.Get(context.TODO(), cronKey, cron)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cron.Status.LastExecutionTimestamp).ToNot(BeNil())
+
+				dv.Status.Phase = cdiv1.Succeeded
+				err = reconciler.client.Update(context.TODO(), dv)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = reconciler.Reconcile(context.TODO(), cronReq)
+				Expect(err).ToNot(HaveOccurred())
+
+				dataSource := &cdiv1.DataSource{}
+				err = reconciler.client.Get(context.TODO(), dataSourceKey(cron), dataSource)
+				Expect(err).ToNot(HaveOccurred())
+
+				sourcePVC := cdiv1.DataVolumeSourcePVC{
+					Namespace: cron.Namespace,
+					Name:      dvName,
+				}
+				Expect(dataSource.Spec.Source.PVC).ToNot(BeNil())
+				Expect(*dataSource.Spec.Source.PVC).To(Equal(sourcePVC))
+
+				err = reconciler.client.Get(context.TODO(), cronKey, cron)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cron.Status.LastImportedPVC).ToNot(BeNil())
+				Expect(*cron.Status.LastImportedPVC).To(Equal(sourcePVC))
+				Expect(cron.Status.LastImportTimestamp).ToNot(BeNil())
+			}
+		},
+			Entry("create DataVolume on AnnSourceDesiredDigest annotation update", false),
+			Entry("update DataImportCron and DataSource on DataVolume Succeeded", true),
+		)
+
+		DescribeTable("Should", func(updateNextCronTime bool) {
```

Comment:
Same thing don't think you need the table. And It with the sequence of events should be sufficient.

Reply:
fixed